kwc-nav.html

```diff
@@ -0,0 +1,232 @@
+<link rel="import" href="../iron-flex-layout/iron-flex-layout.html">
+<link rel="import" href="../kwc-nav-item/kwc-nav-item.html">
+
+<!--
+`kwc-nav`
+Display navigation links within the view header, to select sub-views.
+
+@demo demo/index.html
+-->
+
+<dom-module id="kwc-nav">
+    <template>
+        <style>
+            :host {
+                display: block;
+            }
+            :host .nav-items {
+                @apply --layout-horizontal;
+                @apply --layout-justify-center;
+                font-size: 20px;
+                list-style: none;
+                list-style: none;
+                margin: 4px 0 0 0;
+                padding: 0;
+                position: relative;
+            }
+            :host .caret-container {
+                width: 100%;
+            }
+            :host #caret {
+                display: none;
+                position: relative;
+            }
+            :host #caret::after {
+                background-color: #ffffff;
+                border-radius: 3px 0 0 0;
+                content: "";
+                display: block;
+                height: 12px;
+                left: 0;
+                position: absolute;
+                top: -6px;
+                transform: rotate(45deg);
+                width: 12px;
+            }
+        </style>
+        <nav class="navigation">
+            <ul class="nav-items">
+                <template id="items" is="dom-repeat" items="[[navItems]]" on-dom-change="_moveCaret">
+                    <li>
+                        <kwc-nav-item id="[[item.id]]"
+                                      label="[[item.label]]"
+                                      icon-id="[[item.iconId]]"
+                                      active="[[_isActive(item, activeItemId)]]"
+                                      on-tap="_onTap">
+                                      </kwc-nav-item>
+                    </li>
+                </template>
+            </ul>
+            <template is="dom-if" if="[[hasCaret]]">
+                <div class="caret-container">
+                    <div id="caret"></div>
+                </div>
+            </template>
+        </nav>
+    </template>
+
+    <script>
+        Polymer({
+            is: 'kwc-nav',
+            properties: {
+                /** Flags if navigation should display a caret */
+                hasCaret: {
+                    type: Boolean,
+                    value: false,
+                },
+                /**
+                 * Array of objects representing navigation items. This object
+                 * should follow the format:
+                 * {
+                 *     id: { type: String, required: true },
+                 *     label: { type: String, required: true },
+                 *     iconId: { type: String }
+                 * }
+                 */
+                navItems: {
+                    type: Array,
+                    value: null
```

Comment:
Should this be `null` or an empty Array?

Reply:
I opted to `null` since the `dom-repeat` accepts it. Also I don't want it to be an actual empty array, I want it to start with empty value on the start. In my understanding you can pass an empty array to this element and it's different than passing nothing to it.